src/dp_vnf.c

```diff
@@ -65,6 +113,18 @@ int dp_add_vnf(const uint8_t ul_addr6[DP_VNF_IPV6_ADDR_SIZE], enum dp_vnf_type t
 		return DP_ERROR;
 	}
 
+	if (DP_FAILED(dp_record_vnf_value(vnf))) {
+		if (prefix->ip_type == RTE_ETHER_TYPE_IPV4)
+			DPS_LOG_WARNING("VNF value recording failed", DP_LOG_VNF_TYPE(type), DP_LOG_VNI(vni),
+							DP_LOG_PORTID(port_id), DP_LOG_IPV4(prefix->ipv4), DP_LOG_PREFLEN(prefix_len));
+		else
+			DPS_LOG_WARNING("VNF value recording failed", DP_LOG_VNF_TYPE(type), DP_LOG_VNI(vni),
+							DP_LOG_PORTID(port_id), DP_LOG_IPV6(prefix->ipv6), DP_LOG_PREFLEN(prefix_len));
+		dp_delete_vnf_value(vnf);
```

Comment:
If saving the value failed, then you should not roll it back here, but in the function that failed. Here it has no real effect of course, but in general, that is the pattern everywhere else.

Reply:
it was wrong auto-completed code. It should be rolling back the above step.